### PR TITLE
Added RevenueCat Integration

### LINKED ISSUE #147: RevenueCat integration

### PR DESCRIPTION
**Pull request by planecore**
Closes #147

### CODE REVIEW

File: app/models/purchases/purchases.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  return !!customerInfo.entitlements.active["better-rail-pro"]
+}
+
+export const PurchasesModel = types
+  .model("Purchases")
+  .props({
+    isPro: types.maybe(types.boolean),
+  })
+  .views(() => ({
+    get customerInfo(): Promise<CustomerInfo> {
```
Comment (guytepper) [inline app/models/purchases/purchases.ts:16]: Is the type not inferred automatically?

File: ios/GoogleService-Info.plist [hunk 2]
```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (guytepper) [inline ios/GoogleService-Info.plist:1]: remove, add to gitignore and create a new file 🙏

File: app/screens/planner/planner-screen.tsx [hunk 3]
```diff
@@ -296,6 +297,8 @@ export const PlannerScreen = observer(function PlannerScreen({ navigation }: Pla
           onPress={onGetRoutePress}
           disabled={!routePlan.origin || !routePlan.destination || routePlan.origin.id === routePlan.destination.id}
         />
+
+        <TestPurchases />
```
Comment (guytepper) [inline app/screens/planner/planner-screen.tsx:301]: remove

File: app/models/user/user.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+      return auth().currentUser
+    },
+  }))
+  .actions((self) => ({
+    setCurrentUserId(uid?: string) {
+      self.currentUserId = uid
+    },
+    async afterCreate() {
+      if (!self.currentUser) {
+        auth().signInAnonymously()
```
Comment (guytepper) [inline app/models/user/user.ts:21]: Can you create users only upon purchasing?

**Review by guytepper — CHANGES_REQUESTED** [review]

File: app/models/user/user.ts (hunk 4 above)
Comment (planecore) [reply app/models/user/user.ts:21]: I think it's better to create an anonymous user for everyone. It'll provide us more flexibility in the future and possibly better analytics. It's also free so why not 😄

**Review by guytepper — APPROVED** [review]

**Pull request merged**